### PR TITLE
Added enable board sharing configs for plugin & vertical

### PR DESCRIPTION
**Pull request by cwarnermm**
Documentation for: https://github.com/mattermost/mattermost-server/pull/21333

Notes:
- Added the plugin config setting which was also missing
- Both configs are returned in search results
- Boards end user documentation also updated to provide admin guidance on enabling public sharing based on the Boards deployment type

### CODE REVIEW

File: source/configure/configuration-settings.rst [hunk 1]
```diff
@@ -20,6 +20,7 @@ Mattermost configuration settings are organized into the following categories wi
 - :doc:`Site configuration settings </configure/site-configuration-settings>`
 - :doc:`Authentication configuration settings </configure/authentication-configuration-settings>`
 - :doc:`Plugins configuration settings </configure/plugins-configuration-settings>`
+- :doc:`Product configuration settings <configuration/product-configuration-settings>`
```
Comment (wuwinson) [inline source/configure/configuration-settings.rst:23]:
```suggestion
- :doc:`Product configuration settings </configure/product-configuration-settings>`
```

**Review by wuwinson — APPROVED** [review]
LGTM @cwarnermm! Not sure if line 23 for my suggested change was intentional or not, but if so, then please disregard my suggestion.

**Comment by amyblais** [comment]
Boards as a product is being reverted now (it will ship as a plugin in v7.7) https://github.com/mattermost/mattermost-server/pull/22075. I believe this doc PR is not needed for v7.7.

**Comment by cwarnermm** [comment]
Thanks for the heads-up @amyblais!

**Comment by cwarnermm** [comment]
Note for Carrie: This PR was moved out of the v7.7 docs branch and contains more changed files than needed. Split out relevant source updates into a new PR prior to final reviews/merge.

**Comment by amyblais** [comment]
@cwarnermm This is not needed for v7.8 either as we have reverted Boards back to plugin mode due to an issue found in last week's cloud release.

**Comment by amyblais** [comment]
@cwarnermm This is not needed for v7.9 as we are reverting Boards back to plugin mode due to an issue found in last week's cloud release.

**Comment by cwarnermm** [comment]
Closing since this PR is no longer needed.

**Pull request closed without merging**